src/test/java/teammates/test/cases/ui/pagedata/InstructorCourseEditPageDataTest.java

```diff
@@ -0,0 +1,146 @@
+        instructorList.add(dataBundle.instructors.get("helperOfCourse1"));
+        instructorList.add(dataBundle.instructors.get("instructorNotYetJoinCourse1"));
+        
+        
+        InstructorAttributes currentInstructor = dataBundle.instructors.get("instructor1OfCourse1");
+        
+        int offset = -1;
+        
+        List<String> sectionNames = new ArrayList<String>();
+        sectionNames.add("Section 1");
+        sectionNames.add("Section 2");
+        
+        List<String> feedbackSessionNames = new ArrayList<String>();
+        feedbackSessionNames.add("First feedback session");
+        feedbackSessionNames.add("Second feedback session");
+        feedbackSessionNames.add("Grace Period Session");
+        feedbackSessionNames.add("Closed Session");
+        feedbackSessionNames.add("Empty session");
+        feedbackSessionNames.add("non visible session");
+        
+        InstructorCourseEditPageData pageData = new InstructorCourseEditPageData(account, course, 
+                                                                                 instructorList, 
+                                                                                 currentInstructor, 
+                                                                                 offset, sectionNames, 
+                                                                                 feedbackSessionNames);
+        
+        assertEquals("idOfTypicalCourse1", pageData.getCourse().id);
+        assertEquals(-1, pageData.getInstructorToShowIndex());
+        assertNotNull(pageData.getDeleteCourseButton());
+        assertNotNull(pageData.getAddInstructorButton());
+        
+        assertNotNull(pageData.getInstructorPanelList());
+        assertEquals(instructorList.size(), pageData.getInstructorPanelList().size());
+        
+        CourseEditInstructorPanel panel = pageData.getInstructorPanelList().get(0);
+        assertEquals(4, panel.getPermissionInputGroup1().size());
+        assertEquals(4, panel.getPermissionInputGroup2().size());
+        assertEquals(3, panel.getPermissionInputGroup3().size());
+        assertEquals("idOfInstructor1OfCourse1", panel.getInstructor().googleId);
+        assertNotNull(panel.getDeleteButton());
+        assertNotNull(panel.getEditButton());
+        assertNull(panel.getResendInviteButton());
+        assertEquals(sectionNames.size(), panel.getSectionRows().size());
+        CourseEditSectionRow sectionRow = panel.getSectionRows().get(0);
+        assertEquals(4, sectionRow.getPermissionInputGroup2().size());
+        assertEquals(3, sectionRow.getPermissionInputGroup3().size());
+        assertEquals(feedbackSessionNames.size(), sectionRow.getFeedbackSessions().size());
+        assertNotNull(sectionRow.getToggleSessionLevelInSectionButton());
+        assertFalse(sectionRow.isSectionSpecial());
+        assertEquals((sectionNames.size() - 1) / 3 + 1, sectionRow.getSpecialSections().size());
```

Comment:
Where does `(sectionNames.size() - 1) / 3 + 1` come from?
Maybe a comment or variable to explain?